README.md

```diff
@@ -47,3 +40,95 @@ sudo ./check-payload scan operator \
 IMAGE=some.registry.location/check-payload
 podman  run --privileged -ti -v /:/myroot $IMAGE scan node --root /myroot
 ```
+
+## How it works
+
+`check-payload` gathers container images from OpenShift release payloads or
+container images. The scanner can be used on a RHEL or RHCOS node to scan the
+root image. A node scan runs within a container with the host OS mounted within
+it.
+
+### Container Image and Payload Scans
+
+Container Image and Payload Scans gather images and emit file paths to the
+validation logic. This is done by `podman` mounting the image to the local
+machine, walking the directory tree, and emitting the paths for executables to
+the validation engine.
+
+### Node Scans
+
+RHEL or RHCOS nodes can be scanned with `check-payload scan node`. To gather the
+file input paths the scanner queries for all the RPMs on the system and walks
+the paths within the RPMs finding executables. The list of executable paths are
+then processed by the validation engine.
+
+### Diagram
+
+```mermaid
+flowchart LR
+  subgraph inputs
+  Image --> Scanner
+  Payload --> Scanner
+  Node --> Scanner
+  end
+  Scanner --> Validations
+  Validations --> Printer
+  subgraph results
+  Printer
+  end
+  
+```
+
+### Validations
+
+The validation engine scans executables and golang executables independently. The scanner only scans for ELF executables.
```

Comment:
`s/scans executables and golang executables independently/uses different logic to validate golang and non-golang executables/`.